component/async/kafka/simple/simple.go

```diff
@@ -93,9 +94,19 @@ func (c *consumer) Consume(ctx context.Context) (<-chan async.Message, <-chan er
 	chErr := make(chan error, c.config.Buffer)
 
 	log.Infof("consuming messages from topic '%s' without using consumer group", c.topic)
-	pcs, err := c.partitions()
-	if err != nil {
-		return nil, nil, fmt.Errorf("failed to get partitions: %w", err)
+	var pcs []sarama.PartitionConsumer
+	if c.config.DurationBasedConsumer {
```

Comment:
Would it make sense, instead of having a conditional to use a common interface, and make effectively the switch in the different implementations ?
It might make this code part more straightforward.

Reply:
I get the idea, but how do you see it exactly? 
Adding one interface with a `partitions` method + two new structures were we would copy the required fields from the `consumer` structure (consumer, config, topic) and where we would move this `if c.config.DurationBasedConsumer` into the `Create`?
Is this more straightforward like this?

Reply:
it can be also just a func type, that by default would evaluate to `c.partitions`, and overwritten in our case (in the place where you set the config property to 'true'). From what i see, it s not used somewhere else ?

Reply:
I think it makes perfect sense. Something like this?

Reply:
already looks better to me this way. Thanks!

Reply:
just to highlight, that if we dont use this boolean anywhere else, we might be able to make this assignment already in the 'WithDurationOffset' method and remove the bool. But it s just a detail ...

Reply:
But the partitions are methods so we would need to have access to the consumer receiver by modifying `OptionFunc`.